src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -211,12 +253,28 @@ protected void printSummary() {}
 
   public String produceDts(Compiler compiler) {
     Set<String> provides = new HashSet<>();
-    // TODO: only inspect inputs which were explicitly provided by the user
+    Set<String> transitiveProvides = new HashSet<>();
+    List<String> depgraphRoots = parseDepgraphRoots();
```

Comment:
I think it'd be nice to only optionally take the depgraph files, so that people might still run the tool outside of bazel. E.g. if you have some Google API that's declared as a Closure style externs file.